<?xml version="1.0" encoding="utf-8"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN"
          "DTD/xhtml1-strict.dtd">
<html>
  

  <head>
    
    <title>django.core.exceptions.ValidationError</title>
    <meta name="generator" content="pydoctor 22.2.0"> 
        
    </meta>
    <meta http-equiv="Content-Type" content="text/html;charset=utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=0.75" />
    <link rel="stylesheet" type="text/css" href="bootstrap.min.css" />
    <link rel="stylesheet" type="text/css" href="apidocs.css" />
    <link rel="stylesheet" type="text/css" href="extra.css" />
</head>

  <body>

    

    <nav class="navbar navbar-default">
  
  <div class="container">

    <div class="navbar-header">
      
      <div class="navlinks">
          <span class="navbar-brand">
            django <a href="index.html">API Documentation</a>
          </span>

          <a href="moduleIndex.html">
            Modules
          </a>

          <a href="classIndex.html">
            Classes
          </a>

          <a href="nameIndex.html">
            Names
          </a>
      </div>

    </div>
  </div>
</nav>

    

    <div class="container">

      <div class="page-header">
        <h1 class="class"><code><code><a href="index.html" class="internal-link">django</a></code><wbr></wbr>.<code><a href="django.core.html" class="internal-link" title="django.core">core</a></code><wbr></wbr>.<code><a href="django.core.exceptions.html" class="internal-link" title="django.core.exceptions">exceptions</a></code><wbr></wbr>.<code><a href="django.core.exceptions.ValidationError.html" class="internal-link" title="django.core.exceptions.ValidationError">ValidationError</a></code></code></h1>
        <div id="showPrivate">
          <button class="btn btn-link" onclick="togglePrivate()">Toggle Private API</button>
        </div>
      </div>

      <div class="categoryHeader">
        class documentation
      </div>

      <div class="extrasDocstring">
        <p><code><span class="py-keyword">class</span> <span class="py-defname">ValidationError</span>(Exception):</code></p>
        <p><a href="classIndex.html#django.core.exceptions.ValidationError">View In Hierarchy</a></p>
      </div>

      <div class="moduleDocstring">
        <div>An error while validating data.</div>
      </div>

      <div id="splitTables">
        <table class="children sortable" id="id2797">
  
  
  <tr class="method">
    
    <td>Method</td>
    <td><code><a href="#__eq__" class="internal-link" title="django.core.exceptions.ValidationError.__eq__">__eq__</a></code></td>
    <td><span class="undocumented">Undocumented</span></td>
  </tr><tr class="method">
    
    <td>Method</td>
    <td><code><a href="#__hash__" class="internal-link" title="django.core.exceptions.ValidationError.__hash__">__hash__</a></code></td>
    <td><span class="undocumented">Undocumented</span></td>
  </tr><tr class="method">
    
    <td>Method</td>
    <td><code><a href="#__init__" class="internal-link" title="django.core.exceptions.ValidationError.__init__">__init__</a></code></td>
    <td><span class="undocumented">No summary</span></td>
  </tr><tr class="method">
    
    <td>Method</td>
    <td><code><a href="#__iter__" class="internal-link" title="django.core.exceptions.ValidationError.__iter__">__iter__</a></code></td>
    <td><span class="undocumented">Undocumented</span></td>
  </tr><tr class="method">
    
    <td>Method</td>
    <td><code><a href="#__repr__" class="internal-link" title="django.core.exceptions.ValidationError.__repr__">__repr__</a></code></td>
    <td><span class="undocumented">Undocumented</span></td>
  </tr><tr class="method">
    
    <td>Method</td>
    <td><code><a href="#__str__" class="internal-link" title="django.core.exceptions.ValidationError.__str__">__str__</a></code></td>
    <td><span class="undocumented">Undocumented</span></td>
  </tr><tr class="method">
    
    <td>Method</td>
    <td><code><a href="#update_error_dict" class="internal-link" title="django.core.exceptions.ValidationError.update_error_dict">update​_error​_dict</a></code></td>
    <td><span class="undocumented">Undocumented</span></td>
  </tr><tr class="instancevariable">
    
    <td>Instance Variable</td>
    <td><code><a href="#error_dict" class="internal-link" title="django.core.exceptions.ValidationError.error_dict">error​_dict</a></code></td>
    <td><span class="undocumented">Undocumented</span></td>
  </tr><tr class="property">
    
    <td>Property</td>
    <td><code><a href="#message_dict" class="internal-link" title="django.core.exceptions.ValidationError.message_dict">message​_dict</a></code></td>
    <td><span class="undocumented">Undocumented</span></td>
  </tr><tr class="property">
    
    <td>Property</td>
    <td><code><a href="#messages" class="internal-link" title="django.core.exceptions.ValidationError.messages">messages</a></code></td>
    <td><span class="undocumented">Undocumented</span></td>
  </tr>
</table>
        

          
      </div>

      <div id="childList">

        <div class="basemethod">
  
  
  <a name="django.core.exceptions.ValidationError.__eq__">
    
  </a>
  <a name="__eq__">
    
  </a>
  <div class="functionHeader">
    
    <span class="py-keyword">def</span> <span class="py-defname">__eq__</span>(self, other):
    
  </div>
  <div class="docstring functionBody">
    
    
    <div><p class="undocumented">Undocumented</p></div>
  </div>
</div><div class="basemethod">
  
  
  <a name="django.core.exceptions.ValidationError.__hash__">
    
  </a>
  <a name="__hash__">
    
  </a>
  <div class="functionHeader">
    
    <span class="py-keyword">def</span> <span class="py-defname">__hash__</span>(self):
    
  </div>
  <div class="docstring functionBody">
    
    
    <div><p class="undocumented">Undocumented</p></div>
  </div>
</div><div class="basemethod">
  
  
  <a name="django.core.exceptions.ValidationError.__init__">
    
  </a>
  <a name="__init__">
    
  </a>
  <div class="functionHeader">
    
    <span class="py-keyword">def</span> <span class="py-defname">__init__</span>(self, message, code=None, params=None):
    
  </div>
  <div class="docstring functionBody">
    
    
    <div>The <code><a href="django.core.mail.message.html" class="internal-link" title="django.core.mail.message">message</a></code> argument can be a single error, a list of errors, or a
dictionary that maps field names to lists of errors. What we define as
an "error" can be either a simple string or an instance of
ValidationError with its message attribute set, and what we define as
list or dictionary can be an actual <code><a href="django.views.generic.list.html" class="internal-link" title="django.views.generic.list">list</a></code> or <code>dict</code> or an instance
of ValidationError with its <code>error_list</code> or <code><a href="#error_dict" class="internal-link" title="django.core.exceptions.ValidationError.error_dict">error_dict</a></code> attribute set.</div>
  </div>
</div><div class="basemethod">
  
  
  <a name="django.core.exceptions.ValidationError.__iter__">
    
  </a>
  <a name="__iter__">
    
  </a>
  <div class="functionHeader">
    
    <span class="py-keyword">def</span> <span class="py-defname">__iter__</span>(self):
    
  </div>
  <div class="docstring functionBody">
    
    
    <div><p class="undocumented">Undocumented</p></div>
  </div>
</div><div class="basemethod">
  
  
  <a name="django.core.exceptions.ValidationError.__repr__">
    
  </a>
  <a name="__repr__">
    
  </a>
  <div class="functionHeader">
    
    <span class="py-keyword">def</span> <span class="py-defname">__repr__</span>(self):
    
  </div>
  <div class="docstring functionBody">
    
    
    <div><p class="undocumented">Undocumented</p></div>
  </div>
</div><div class="basemethod">
  
  
  <a name="django.core.exceptions.ValidationError.__str__">
    
  </a>
  <a name="__str__">
    
  </a>
  <div class="functionHeader">
    
    <span class="py-keyword">def</span> <span class="py-defname">__str__</span>(self):
    
  </div>
  <div class="docstring functionBody">
    
    
    <div><p class="undocumented">Undocumented</p></div>
  </div>
</div><div class="basemethod">
  
  
  <a name="django.core.exceptions.ValidationError.update_error_dict">
    
  </a>
  <a name="update_error_dict">
    
  </a>
  <div class="functionHeader">
    
    <span class="py-keyword">def</span> <span class="py-defname">update_error_dict</span>(self, error_dict):
    
  </div>
  <div class="docstring functionBody">
    
    
    <div><p class="undocumented">Undocumented</p></div>
  </div>
</div><div class="baseinstancevariable">
  
  
  <a name="django.core.exceptions.ValidationError.error_dict">
    
  </a>
  <a name="error_dict">
    
  </a>
  <div class="functionHeader">
    
    <span class="py-defname">error_dict</span>: <code>dict</code> =
    
  </div>
  <div class="functionBody">
    
    
    <div><p class="undocumented">Undocumented</p></div>
    
  </div>
</div><div class="baseproperty">
  
  
  <a name="django.core.exceptions.ValidationError.message_dict">
    
  </a>
  <a name="message_dict">
    
  </a>
  <div class="functionHeader">
    @property<br />
    <span class="py-defname">message_dict</span> =
    
  </div>
  <div class="functionBody">
    
    
    <div><p class="undocumented">Undocumented</p></div>
    
  </div>
</div><div class="baseproperty">
  
  
  <a name="django.core.exceptions.ValidationError.messages">
    
  </a>
  <a name="messages">
    
  </a>
  <div class="functionHeader">
    @property<br />
    <span class="py-defname">messages</span> =
    
  </div>
  <div class="functionBody">
    
    
    <div><p class="undocumented">Undocumented</p></div>
    
  </div>
</div>

      </div>
    </div>

    <footer class="navbar navbar-default">
  
  <div class="container">
    <a href="index.html">API Documentation</a> for django,
  generated by <a href="https://github.com/twisted/pydoctor/">pydoctor</a>
    22.2.0 at 2022-02-23 04:30:39.
  </div>
</footer>

    <script src="pydoctor.js" type="text/javascript"></script>

  </body>
</html>